test/e2e/framework/wait.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/format"
+	gomegatypes "github.com/onsi/gomega/types"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// GetFunc is a function which retrieves a certain object.
+type GetFunc[T any] func(ctx context.Context) (T, error)
```

Comment:
heh, so this is the first Generic in the codebase? 😄

Reply:
It's still in the race, but https://github.com/kubernetes/kubernetes/pull/112377 will probably be faster :grin: